artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/impl/PagingStoreImpl.java

```diff
@@ -1413,8 +1500,18 @@ private void installPageTransaction(final Transaction tx, final RouteContextList
       return;
    }
 
+   @Override
+   public boolean hasPendingIO() {
+      return timedWriter != null && timedWriter.hasPendingIO();
+   }
+
+   public PageTimedWriter getPageTimedWriter() {
+      return timedWriter;
+   }
+
    @Override
    public void destroy() throws Exception {
+      this.timedWriter.stop();
```

Comment:
Since timedWriter can be null, would seem like this should have a null check.

The "this." is inconsistent with the method above.